cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -1217,6 +1228,11 @@ def _create_snapshot(self, snapshot):
 
         :param snapshot: Snapshot object
         """
+        backend = None
+        key = "__cinder_internal_backend"
+        if ('metadata' in snapshot and snapshot['metadata'] and
+                key in snapshot['metadata']):
+            backend = snapshot['metadata'][key]
```

Comment:
Just a thought I had:
```suggestion
        if snapshot.get('metadata'):
            backend = snapshot['metadata'].get(key)
```